modules/internal/Bytes.chpl

```diff
@@ -705,26 +796,46 @@ module Bytes {
     }
 
     /*
-      Returns a new :record:`bytes` object, which is the concatenation of all of
+      Returns a new :record:`bytes`, which is the concatenation of all of
       the :record:`bytes` objects passed in with the contents of the method
       receiver inserted between them.
+
+      .. code-block:: chapel
+
+          var x = b"|".join(b"a",b"10",b"d");
+          writeln(x); // prints: "a|10|d"
+
+      :arg S: Object(s) to be joined
+      :type S: :record:`bytes`
+
+      :returns: A :record:`bytes`
     */
     inline proc join(const ref S: bytes ...) : bytes {
       return _join(S);
     }
 
     /*
-      Same as the varargs version, but with a homogeneous tuple of bytes.
+      Returns a new :record:`bytes`, which is the concatenation of all of
+      the :record:`bytes` objects passed in with the contents of the method
+      receiver inserted between them.
+
+      .. code-block:: chapel
+
+          var x = b"|".join((b"a",b"10",b"d"));
+          writeln(x); // prints: "a|10|d"
+
+      :arg S: Object(s) to be joined
```

Comment:
I'm not sure I'd refer to a `bytes` as an object. I'm not sure that the arg/type distinction is helping here.